src/app/components/resources/resources.component.ts

```diff
@@ -13,36 +18,100 @@ import { LanguageService } from '../../service/language.service';
 export class ResourcesComponent implements OnInit {
   resources: Resource[];
   languages: Language[];
+  resourceTypes: ResourceType[];
+  systems: System[];
 
   showInstructions = false;
+  showFilters = false;
   loadingResources = false;
   loadingLanguages = false;
   errorMessage: string;
+  filters: Filters;
+  totalFilters: number;
+  sortOrder: string;
 
   constructor(
     private resourceService: ResourceService,
     private languageService: LanguageService,
     private modalService: NgbModal,
+    protected systemService: SystemService,
+    protected resourceTypeService: ResourceTypeService,
   ) {}
 
   ngOnInit(): void {
     this.loadResources();
     this.loadLanguages();
+    this.loadFilters();
   }
 
-  loadResources(): void {
-    this.loadingResources = true;
+  blankFilters(): Filters {
+    return {
+      type: [],
+      system: [],
+      other: [],
+    };
+  }
 
+  loadResources(): void {
     this.resourceService
       .getResources(
         'latest-drafts-translations,pages,custom-manifests,tips,attachments,variants',
       )
       .then((resources) => {
-        this.resources = resources;
+        const localFilters = localStorage.getItem('filters');
+        this.sortOrder = localStorage.getItem('sortOrder');
+        this.filters = localFilters
+          ? JSON.parse(localFilters)
+          : this.blankFilters();
+        this.totalFilters = Object.keys(this.filters).reduce(
+          (acc, item) => acc + this.filters[item].length,
+          0,
+        );
+
+        this.resources = this.filters
```

Comment:
This logic is kind of complex, so it would be a good candidate for adding tests for. Or at least moving the filtering logic into a helper function and adding tests for the helper.